### PR TITLE
Add sample codes using assert_unary_iteration() and assert_circuit_equivalent_to_output_qubit_state()

### PR DESCRIPTION
**Pull request by junnaka51**
In this PR I add 4 sample code files which use the assert methods `assert_unary_iteration()` and `assert_circuit_equivalent_to_output_qubit_state()`. 

I believe these files will help users understand when and how they can use these assert methods.

### CODE REVIEW

File: sample/adder.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            bitstring_b = ("0" * 5 + bin(decimal_b)[2:])[-5:]
+            decimal_c = decimal_a + decimal_b
+            if decimal_c > 31:
+                continue
+            bitstring_c = ("0" * 5 + bin(decimal_c)[2:])[-5:]
+            # print(decimal_a, decimal_b, decimal_c)
+
+            assert_circuit_equivalent_to_output_qubit_state(
+                circuit=circuit,
+                # input_reg = reg. for b + reg. for a
```
Comment (mitz1012) [inline sample/adder.py:29]: It may be better to place the comment similarly to line 31.

File: sample/adder.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    for decimal_a in range(2**5):
+        for decimal_b in range(2**5):
+            bitstring_a = ("0" * 5 + bin(decimal_a)[2:])[-5:]
+            bitstring_b = ("0" * 5 + bin(decimal_b)[2:])[-5:]
+            decimal_c = decimal_a + decimal_b
+            if decimal_c > 31:
+                continue
+            bitstring_c = ("0" * 5 + bin(decimal_c)[2:])[-5:]
+            # print(decimal_a, decimal_b, decimal_c)
```
Comment (mitz1012) [inline sample/adder.py:25]: If unnecessary, please delete this line.

File: sample/adder.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+def test_5_bit_adder(circuit):
+    """a + b = c"""
+
+    for decimal_a in range(2**5):
+        for decimal_b in range(2**5):
+            bitstring_a = ("0" * 5 + bin(decimal_a)[2:])[-5:]
+            bitstring_b = ("0" * 5 + bin(decimal_b)[2:])[-5:]
+            decimal_c = decimal_a + decimal_b
+            if decimal_c > 31:
+                continue
```
Comment (mitz1012) [inline sample/adder.py:23]: `continue` can be replace to the line before line 19.

**Review by mitz1012 — COMMENTED** [review]
Thank you very much for preparing sample programs. There are basically no problems with the pull requests. Sorry for being so fussy, but please check my comments.

File: sample/adder.py (hunk 1 above)
Comment (junnaka51) [reply sample/adder.py]: Done in https://github.com/QuantestPy/quantestpy/pull/198/commits/f7e800f85e0e867a96482122358e2e5132b64a92

File: sample/adder.py (hunk 2 above)
Comment (junnaka51) [reply sample/adder.py]: Done in https://github.com/QuantestPy/quantestpy/pull/198/commits/f7e800f85e0e867a96482122358e2e5132b64a92

File: sample/adder.py (hunk 3 above)
Comment (junnaka51) [reply sample/adder.py:21]: Done in https://github.com/QuantestPy/quantestpy/pull/198/commits/f7e800f85e0e867a96482122358e2e5132b64a92

**Review by mitz1012 — APPROVED** [review]

**Pull request merged**